includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -311,16 +311,26 @@ public function get_product_data() {
 
 		global $post;
 
-		$product = wc_get_product( $post->ID );
+		$product      = wc_get_product( $post->ID );
+		$total_amount = WC_Stripe_Helper::get_stripe_amount( $product->get_price() );
 
 		if ( 'variable' === $product->get_type() ) {
 			$attributes = wc_clean( wp_unslash( $_GET ) );
 
+			if ( empty( $attributes ) ) {
```

Comment:
I've looked into this and haven't found a case where `$_GET` will actually be used. The `get_product_data` method is only used on the `scripts` method which is used to expose data to the frontend via `add_action( 'wp_enqueue_scripts', [ $this, 'scripts' ] )`.

I'll see where this was introduced to see if it's really necessary otherwise I'll remove it.

Reply:
👍 ICYMI setting a `?attribute_size=large` URL param prefills that attribute value.

Reply:
@dechov The code was introduced a year ago on this PR https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1149 by @DanReyLop 

It seems to me that previously we were able to select a variation via `GET`? It would be nice some guidance here since I don't see if this is case anymore.

Reply:
> 👍 ICYMI setting a ?attribute_size=large URL param prefills that attribute value.

yeah, I was assuming something like that would be possible. In any case what do you suggest here in order to keep that functionality and the fix I'm doing here?

Reply:
actually the query string is something like this: `?attribute_pa_size=extra-large` in order to work properly.

Reply:
@dechov would this be good to you: Ask if is there at least one `$_GET` variable starting with `attribute_` to use `$_GET` variables otherwise to use the `$product->get_default_attributes()`? Something like:

```php
$attributes = at_least_one('attribute_', $_GET) ? wc_clean( wp_unslash( $_GET ) ) : $product->get_default_attributes();
```
```

Reply:
There is WooCommerce core functionality to pre-fill the attribute selection by passing a query param – not sure about the `pa_` as only `attribute_size=large` is working for me. I don't know if this feature is documented, and I only discovered it by looking at that code (introduced in the PR you linked) which attempts compatibility with it by reading from `$_GET` so that those pre-filled attributes can be used to prepare the payment request total as well.

I think the overall goal here is to match the WC core behavior (i.e. show in the payment request flow the same price that appears for the currently selected variation on the page), which appears to fall back to the default attribute if (and only if) a query parameter isn't passed.

Reply:
> There is WooCommerce core functionality to pre-fill the attribute selection by passing a query param
Yeah, I just discovered that.

> I think the overall goal here is to match the WC core behavior
That's what I'm working now. I'll update my PR shortly